tests/trustless_gateway_car_test.go

```diff
@@ -579,6 +579,107 @@ func TestTrustlessCarEntityBytes(t *testing.T) {
+			Response: Expect().
+				Status(200).
+				Headers(
+					Header("Content-Type").Contains("application/vnd.ipld.car"),
+					Header("Content-Type").Contains("order=dfs"),
+					Header("Content-Type").Contains("dups=y"),
+				).
+				Body(
+					IsCar().
+						IgnoreRoots().
+						HasBlocks(flattenStrings(t,
+							dirWithDuplicateFiles.MustGetCid(),
+							dirWithDuplicateFiles.MustGetCid("ascii.txt"), // ascii.txt = ascii-copy.txt
+							dirWithDuplicateFiles.MustGetCid("ascii-copy.txt"),
+							dirWithDuplicateFiles.MustGetCid("hello.txt"),
+							dirWithDuplicateFiles.MustGetCid("multiblock.txt"),
+							dirWithDuplicateFiles.MustGetChildrenCids("multiblock.txt"),
+						)...).
+						Exactly().
+						InThatOrder(),
+				),
+		},
+		{
+			Name: "GET CAR with dups=n of UnixFS Directory With Duplicate Files",
+			Hint: `
+				The response MUST NOT contain duplicate blocks.In this test, a directory that
+				contains duplicate files is requested. The blocks corresponding to the duplicate
+				files must be returned only ONCE.
+			`,
+			Request: Request().
+				Path("/ipfs/{{cid}}", dirWithDuplicateFiles.MustGetCid()).
+				Header("Accept", "application/vnd.ipld.car; version=1; order=dfs; dups=n"),
+			Response: Expect().
+				Status(200).
+				Headers(
+					Header("Content-Type").Contains("application/vnd.ipld.car"),
+					Header("Content-Type").Contains("order=dfs"),
+					Header("Content-Type").Contains("dups=n"),
+				).
+				Body(
+					IsCar().
+						IgnoreRoots().
+						HasBlocks(flattenStrings(t,
+							dirWithDuplicateFiles.MustGetCid(),
+							dirWithDuplicateFiles.MustGetCid("ascii.txt"), // ascii.txt = ascii-copy.txt
+							dirWithDuplicateFiles.MustGetCid("hello.txt"),
+							dirWithDuplicateFiles.MustGetCid("multiblock.txt"),
+							dirWithDuplicateFiles.MustGetChildrenCids("multiblock.txt"),
+						)...).
+						Exactly().
+						InThatOrder(),
+				),
+		},
+		{
+			Name: "GET CAR with order=unk of UnixFS Directors",
+			Hint: `
+				The response with order=unk MUST contain all the blocks required to construct
+				the requested CID. However, it can be in any order and duplicates MAY occur.
+			`,
+			Request: Request().
+				Path("/ipfs/{{cid}}", dirWithDuplicateFiles.MustGetCid()).
+				Header("Accept", "application/vnd.ipld.car; version=1; order=unk"),
+			Response: Expect().
+				Status(200).
+				Headers(
+					Header("Content-Type").Contains("application/vnd.ipld.car"),
+					Header("Content-Type").Contains("order="),
+				).
+				Body(
+					IsCar().
+						IgnoreRoots().
+						HasBlocks(flattenStrings(t,
+							dirWithDuplicateFiles.MustGetCid(),
+							dirWithDuplicateFiles.MustGetCid("ascii.txt"),
+							dirWithDuplicateFiles.MustGetCid("ascii-copy.txt"),
+							dirWithDuplicateFiles.MustGetCid("hello.txt"),
+							dirWithDuplicateFiles.MustGetCid("multiblock.txt"),
+							dirWithDuplicateFiles.MustGetChildrenCids("multiblock.txt"),
+						)...),
```

Comment:
golang's splicing is quite restricted :/ I'd like to get rid of `flattenStrings`, which is too low level for these test definitions; what about we use:

```suggestion
						HasBlock(dirWithDuplicateFiles.MustGetCid()).
						HasBlock(dirWithDuplicateFiles.MustGetCid("ascii.txt")).
						HasBlock(dirWithDuplicateFiles.MustGetCid("ascii-copy.txt")).
						HasBlock(dirWithDuplicateFiles.MustGetCid("hello.txt")).
						HasBlock(dirWithDuplicateFiles.MustGetCid("multiblock.txt")).
						HasBlocks(dirWithDuplicateFiles.MustGetChildrenCids("multiblock.txt")).
```

?